nodes/generators_extended/spiral_mk2.py

```diff
@@ -0,0 +1,655 @@
+        N        : the curve resolution of one turn
+        scale    : overall scale of the curve
+        height   : the height of the spiral along z
+        phase    : phase the spiral around its center
+        flip     : flip the spiral direction (default is CLOCKWISE)
+    '''
+
+    eR, iR, exponent, turns, N, scale, height, phase, flip = settings
+
+    sign = -1 if flip else 1  # flip direction ?
+
+    max_phi = 2 * pi * turns * sign
+
+    # derive eR based on iR and height (the main parameters)
+    # eR = [iR - (H/2)^2/iR]/2 ::: H = 2 * sqrt(2*iR*eR - iR*iR)
+    eR = 0.5 * (iR + 0.25 * height * height / iR)
+    eR2 = eR * eR  # cached for performance
+    dR = eR - iR  # cached for performance
+
+    N = N * turns  # total number of points in the spiral
+
+    es = prepareExponentialSettings(2, exponent + 1e-5)  # used for easing
+
+    verts = []
+    norms = []
+    add_vert = verts.append
+    add_norm = norms.append
+    for n in range(N + 1):
+        t = n / N  # t : [0, 1]
+        phi = max_phi * t + phase
+        a = ExponentialEaseInOut(t, es)  # ease theta variation
+        theta = -pi / 2 + pi * a
+        h = 0.5 * height * sin(theta)  # [-H/2, +H/2]
+        r = sqrt(eR2 - h * h) - dR  # [0 -> iR -> 0]
+        x = r * cos(phi) * scale
+        y = r * sin(phi) * scale
+        z = h * scale
+        add_vert([x, y, z])
+
+    edges = get_edge_list(N)
+
+    return verts, edges, norms
+
+
+def make_cornu_spiral(settings):
+    '''
+        L     : length
+        N     : resolution
+        S     : scale
+        M     :
+
+        x(t) = s * Integral(0,t) { cos(pi*u*u/2) du }
+        y(t) = s * Integral(0,t) { sin(pi*u*u/2) du }
+
+        TODO : refine the math (smoother curve, adaptive res, faster computation)
+    '''
+
+    eR, iR, exponent, turns, N, scale, height, phase, flip = settings
+
+    sign = -1 if flip else 1  # flip direction ?
+
+    N = N * turns  # total number of points in the spiral
+    L = iR * turns  # length
+    S = eR * scale  # overall scale
+
+    es = prepareExponentialSettings(2, exponent + 1e-5)  # used for easing
+
+    verts1 = []  # pozitive spiral verts
+    verts2 = []  # nagative spiral verts
+    norms = []
+    add_vert1 = verts1.append
+    add_vert2 = verts2.append
+    add_norm = norms.append
+    l1 = 0
+    x = 0
+    y = 0
+    for n in range(N + 1):
+        t = n / N  # t = [0,1]
+
+        a = QuadraticEaseOut(t)
+        # a = ExponentialEaseOut(t, es)
+
+        l = L * a  # l = [0, +L]
+
+        r = x * x + y * y
+        # print("r=", r)
+        # M = 100 + int(300 * pow(r, exponent)) # integral steps
+        M = 100 + int(100 * a)  # integral steps
+        l2 = l
+
+        # integral from l1 to l2
+        u = l1
+        du = (l2 - l1) / M
+        for m in range(M + 1):
+            u = u + du  # u = [l1, l2]
+            phi = u * u * pi / 2
+            x = x + cos(phi) * du
+            y = y + sin(phi) * du
+        l1 = l2
```

Comment:
This quite magical code if someone have to fix something he can have difficulties with code interpretation. More meaningful names of variables are appreciated.

Reply:
> This quite magical code if someone have to fix something he can have difficulties with code interpretation. More meaningful names of variables are appreciated.

agree. this spiral is mostly experimental however.  did my best to optimize for speed since this usually involves integrals and left it in a state where it was mostly working. I’ll make another pass to improve it in the future, and will probably try to come up with better notation.